src/gen9_vdenc.c

```diff
@@ -3670,13 +3670,13 @@ gen9_vdenc_avc_encode_picture(VADriverContextP ctx,
         gen9_vdenc_mfx_vdenc_pipeline(ctx, encode_state, encoder_context);
         gen9_vdenc_read_status(ctx, encoder_context);
 
-        intel_batchbuffer_end_atomic(batch);
-        intel_batchbuffer_flush(batch);
-
         vdenc_context->brc_initted = 1;
         vdenc_context->brc_need_reset = 0;
     }
 
+    intel_batchbuffer_end_atomic(batch);
+    intel_batchbuffer_flush(batch);
+
     return VA_STATUS_SUCCESS;
 }
 
```

Comment:
Yakui's fix is trying to use one ioctl call but with a big batchbuffer to replace two ioctl calls with small batchbuffers. A big batchbuffer consumes more GPU resource and might delay the execution of other batchbuffers from other applications . On the other hand, I think the increase in performance is trivial because the commands for the 2nd pass is also submitted to kernel for the case of big batchbuffer.  Note we can not decide the execution of the 2nd pass in CPU side,  otherwise CPU has to check the result of the first batchbuffer , which implies a sync between CPU and GPU and will impact the performance a lot.